### PR TITLE
Multiple config templates

### PR DESCRIPTION
**Pull request by ccorcoveanu**

### CODE REVIEW

File: erdpy-up.py [hunk 1]
```diff
@@ -230,6 +234,29 @@ def get_profile_file():
     return os.path.expanduser(file)
 
 
+def upgrade_erdpy_config():
```
Comment (andreibancioiu) [inline erdpy-up.py:237]: :+1:

File: erdpy/config.py [hunk 2]
```diff
@@ -51,14 +51,48 @@ def get_dependency_url(key: str, tag: str, platform: str) -> str:
...
-    data = read_file()
+    data = get_active()
     return data.get(name, get_defaults()[name])
 
 
 def set_value(name: str, value: Any):
     _guard_valid_name(name)
     data = read_file()
-    data[name] = value
+    data["configurations"][data["active"]][name] = value
```
Comment (andreibancioiu) [inline erdpy/config.py:61]: Perhaps use `theDictionary.get(key, fallback)` pattern instead (currently the build on Github Actions fails, somehow related to this).

Comment (ccorcoveanu) [reply erdpy/config.py]: done, looks decent but not really happy, if you have a suggestion for prettier code, I'm open 🥇

**Review by andreibancioiu — DISMISSED** [review]

File: erdpy/cli_config.py [hunk 3]
```diff
@@ -24,6 +24,18 @@ def setup_parser(subparsers: Any) -> Any:
     sub.add_argument("value", help="the new value")
     sub.set_defaults(func=set_value)
 
+    sub = cli_shared.add_command_subparser(subparsers, "config", "new", "Creates a new configuration.")
+    _add_name_arg(sub)
+    sub.add_argument("--template", required=False, help="template from which to create the new config")
+    sub.set_defaults(func=new_config)
+
+    sub = cli_shared.add_command_subparser(subparsers, "config", "switch", "Switch to a new defined config")
```
Comment (camilbancioiu) [inline erdpy/cli_config.py:32]: Suggest "Switch to a different config"

File: erdpy/config.py [hunk 4]
```diff
@@ -51,14 +51,54 @@ def get_dependency_url(key: str, tag: str, platform: str) -> str:
...
+
+
+def set_active(name: str):
+    data = read_file()
+    _guard_valid_config_name(data, name)
+    data["active"] = name
+    write_file(data)
+
+
+def create_new(name: str, template: str):
```
Comment (camilbancioiu) [inline erdpy/config.py:83]: Suggest renaming function to `create_new_config()`.

**Review by camilbancioiu — DISMISSED** [review]

File: erdpy/cli_config.py (hunk 3 above)
Comment (ccorcoveanu) [reply erdpy/cli_config.py:32]: Done

File: erdpy/config.py (hunk 4 above)
Comment (ccorcoveanu) [reply erdpy/config.py:83]: Done

**Review by andreibancioiu — APPROVED** [review]

**Review by camilbancioiu — APPROVED** [review]

**Comment by eddienuta** [comment]
Awesome job guys, as always!!

**Pull request merged**